File: teacher-view.php
<!DOCTYPE html>

<?php
use Symfony\Component\Console\Helper\TableCell;
use Symfony\Component\Console\Helper\TableRows;

require 'view-functionality.php';
include 'connection.php';

//if (isset($_COOKIE['token'])) {

  $pressed = false;
  if (isset($_POST['printout-button'])) {
    // Redirect the user to printout.php
    header("Location: all-printout.php", true);
    exit();
  }


  if (isset($_POST['new-student-button'])) {
    newStudentButton();
  }

  $password = "token";
  if (isset($_POST['token-button'])) {
    unset($_POST['token-button']);
    $token = $_POST['token-text'];

    $pressed = true;
    if (strlen(trim($token)) == 0) {
      echo '<script>alert("Token is empty");</script>';
    } else {
      if (!preg_match('/^[a-zA-Z0-9]+$/', $token)) {
        echo '<script>alert("Token contains invalid characters");</script>';
      } else {
        if (strcmp($token, $password) == 0) {
          echo '<script>alert("Token is correct");</script>';
        } else {
          echo '<script>alert("Token is incorrect");</script>';
        }
      }
    }
    
  }
  

//}


?>

<html lang="en">

<head>
  <meta charset="utf-8" />
  <title id="title">Teacher View</title>
  <link rel="stylesheet" href="teacher-view.css" />
</head>

<body>
  <form>
    <label for="class">Class</label>
    <select name="class" id="class">
      <option value="1">1</option>
      <option value="2">2</option>
      <option value="3">3</option>
      <option value="4">4</option>
      <option value="5">5</option>
      <option value="6">6</option>
      <option value="7">7</option>
      <option value="8">8</option>
      <option value="9">9</option>
      <option value="10">10</option>
      <option value="11">11</option>
      <option value="12">12</option>
    </select>
    <label for="term">Term</label>
    <select name="term" id="term">
      <option value="1">1</option>
      <option value="2">2</option>
      <option value="3">3</option>
    </select>
    <label for="year">Year</label>
    <select name="year" id="year">
      <option value="2021">2021</option>
      <option value="2022">2022</option>
      <option value="2023">2023</option>
      <option value="2024">2024</option>
      <option value="2025">2025</option>
      <option value="2026">2026</option>
      <option value="2027">2027</option>
      <option value="2028">2028</option>
      <option value="2029">2029</option>
      <option value="2030">2030</option>
    </select>
    <br /><br />
  </form>

  <form method="POST">
    <div id="control-center">
      <textarea id="new-student" cols="30" rows="1"></textarea>
      <button type="submit" id="search-student-button">Search Student</button>

      <textarea id="token-text" cols="40" rows="1" name="token-text"></textarea>
      <button type="submit" id="token-button" name="token-button">Enter Token</button>
      <button type="submit" id="new-student-button" name="new-student-button">New Student</button>
      <button type="submit" id="printout-button" name="printout-button">View Print Out</button>
    </div>
  </form>
  <?php
		$sql = "SELECT * FROM GradeTbl";
		$result = mysqli_query($con, $sql);
		$queryResults = mysqli_num_rows($result);
    $curr = 1;
		echo "<table width='100%'>";
		if ($queryResults > 0) {
			echo "<tr>";
				echo "<th class = 'rotate-text'>Class</th>";
				echo "<th class = 'rotate-text'>Name</th>";
				echo "<th class = 'rotate-text'>Position</th>";
				echo "<th class = 'rotate-text'>Best of Five plus Eng</th>";
				echo "<th class = 'rotate-text'>Running Average</th>";
				echo "<th class = 'rotate-text'>POINTS</th>";
				echo "<th class = 'rotate-text'>English</th>";
				echo "<th class = 'rotate-text'>Physics</th>";
				echo "<th class = 'rotate-text'>Chemistry</th>";
				echo "<th class = 'rotate-text'>Biology</th>";
				echo "<th class = 'rotate-text'>Maths</th>";
				echo "<th class = 'rotate-text'>Social Studies</th>";
				echo "<th class = 'rotate-text'>History</th>";
				echo "<th class = 'rotate-text'>Geography</th>";
				echo "<th class = 'rotate-text'>Agriculture</th>";
				echo "<th class = 'rotate-text'>Bible Knowledge</th>";
				echo "<th class = 'rotate-text'>Chichewa</th>";
				echo "<th class = 'rotate-text'>French</th>";
				echo "<th class = 'rotate-text'>Computer Studies</th>";
				echo "<th class = 'rotate-text'>Life Skills</th>";
				echo "</tr>";
			while($row = mysqli_fetch_assoc($result)) {
				echo "<tr>";
				echo "<td>".$row['ClassID']."</td>";
				echo "<th name=\"teacher-edit.php?ID=$curr\">"."<a href='teacher-edit.php?ID=".$row['StudentID']."'>".$row['FirstName']." ".$row['LastName']."</a></th>";
        $curr = $curr + 1;
				echo "<td>".$row['StudentPosition']."</td>";
				echo "<td>".$row['BOFPE']."</td>";
				echo "<td>".$row['RunningAverage']."</td>";
				echo "<td>".$row['Points']."</td>";
				echo "<td>".$row['English']."</td>";
				echo "<td>".$row['Physics']."</td>";
				echo "<td>".$row['Chemistry']."</td>";
				echo "<td>".$row['Biology']."</td>";
				echo "<td>".$row['Math']."</td>";
				echo "<td>".$row['SocialStudies']."</td>";
				echo "<td>".$row['History']."</td>";
				echo "<td>".$row['Geography']."</td>";
				echo "<td>".$row['Agriculture']."</td>";
				echo "<td>".$row['BibleKnowledge']."</td>";
				echo "<td>".$row['Chichewa']."</td>";
				echo "<td>".$row['French']."</td>";
				echo "<td>".$row['ComputerStudies']."</td>";
				echo "<td>".$row['LifeSkills']."</td>";
				echo "</tr>";
			}
		}
		echo "</table>";
?>

	
</body>

</html>

File: testSite/teacher-view.py
#!/usr/bin/python3
import cgi
import connection 

def printTable():
    
    cursor = connection.connect()
    sql = "SELECT * FROM GradeTbl"
    cursor.execute(sql)
    result = cursor.fetchall()
    queryResults = cursor.rowcount
    curr = 1
    print("<table width=100%>")
    if(queryResults > 0):
        print ("<tr>")
        print("<th class = 'rotate-text'>Class</th>")
        print( "<th class = 'rotate-text'>Name</th>")
        print( "<th class = 'rotate-text'>Position</th>")
        print( "<th class = 'rotate-text'>Best of Five plus Eng</th>")
        print( "<th class = 'rotate-text'>Running Average</th>")
        print( "<th class = 'rotate-text'>POINTS</th>")
        print( "<th class = 'rotate-text'>English</th>")
        print( "<th class = 'rotate-text'>Physics</th>")
        print( "<th class = 'rotate-text'>Chemistry</th>")
        print( "<th class = 'rotate-text'>Biology</th>")
        print( "<th class = 'rotate-text'>Maths</th>")
        print( "<th class = 'rotate-text'>Social Studies</th>")
        print( "<th class = 'rotate-text'>History</th>")
        print( "<th class = 'rotate-text'>Geography</th>")
        print( "<th class = 'rotate-text'>Agriculture</th>")
        print( "<th class = 'rotate-text'>Bible Knowledge</th>")
        print( "<th class = 'rotate-text'>Chichewa</th>")
        print( "<th class = 'rotate-text'>French</th>")
        print( "<th class = 'rotate-text'>Computer Studies</th>")
        print( "<th class = 'rotate-text'>Life Skills</th>")
        print( "</tr>")
        for row in result:
            print("<tr>")
            
            classID = row[1]
            if(classID is None):
                classID = "&nbsp"
            print("<td>" + classID + "</td>")
            
            #get values from table
            studentID = str(row[0])
            firstName = str(row[2])
            lastName = str(row[3])    
          
            # print("<th><a href='teacher-edit.py?ID='" + studentID + ">" + firstName + " " + lastName+ "</a></th>")
            print("<th><a href=teacher-edit.py?ID=" + studentID + ">" + firstName + " " + lastName+ "</a></th>")
            curr += 1
            
            studentPosition = row[4]
            if(studentPosition is None):
                studentPosition = "&nbsp"
            print("<td>" + studentPosition + "</td>")
            
            bofpe = row[5]
            if(bofpe is None):
                bofpe = "&nbsp"
            print("<td>" + bofpe + "</td>")
            
            running = row[6]
            if(running is None):
                running = "&nbsp"
            print("<td>" + running + "</td>")
            
            points = row[7]
            if(points is None):
                points = "&nbsp"
            print("<td>" + points + "</td>")
        
            english = row[8]
            if(english is None):
                english = "&nbsp"
            print("<td>" + english + "</td>")
            
            physics = row[9]
            if(physics is None):
                physics = "&nbsp"
            print("<td>" + physics + "</td>")
            
            chemistry = row[10]
            if(chemistry is None):
                chemistry = "&nbsp"
            print("<td>" + chemistry + "</td>")
            
            biology = row[11]
            if(biology is None):
                biology = "&nbsp"
            print("<td>" + biology + "</td>")
            
            maths = row[12]
            if(maths is None):
                maths = "&nbsp"
            print("<td>" + maths + "</td>")
            
            socialStudies = row[13]
            if(socialStudies is None):
                socialStudies = "&nbsp"
            print("<td>" + socialStudies + "</td>")
            
            history = row[14]
            if(history is None):
                history = "&nbsp"
            print("<td>" + history + "</td>")
            
            geography = row[15]
            if(geography is None):
                geography = "&nbsp"
            print("<td>" + geography + "</td>")
            
            agriculture = row[16]
            if(agriculture is None):
                agriculture = "&nbsp"
            print("<td>" + agriculture + "</td>")

            bibleKnowledge = row[17]
            if(bibleKnowledge is None):
                bibleKnowledge = "&nbsp"
            print("<td>" + bibleKnowledge + "</td>")
            
            chichewa = row[18]
            if(chichewa is None):
                chichewa = "&nbsp"
            print("<td>" + chichewa + "</td>")
            
            french = row[19]
            if(french is None):
                french = "&nbsp"
            print("<td>" + french + "</td>")
            
            computerStudies = row[20]
            if(computerStudies is None):
                computerStudies = "&nbsp"
            print("<td>" + computerStudies + "</td>")
            
            lifeSkills = row[21]
            if(lifeSkills is None):
                lifeSkills = "&nbsp"
            print("<td>" + lifeSkills + "</td>")
            
            print("</tr>")
    print("</table>")
    cursor.close()
    
print("Content-type: text/html")
print("")
print(open('teacher-view.html').read())
printTable()
